src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -1628,6 +1628,13 @@ MulticopterPositionControl::control_offboard()
 
 			_hold_offboard_z = false;
 
+			/* disable position control as it is not enabled */
+			if(_control_mode.flag_control_position_enabled){
+				_run_pos_control = true;
+			} else {
+				_run_pos_control = false;
```

Comment:
This is slightly scary. Have you tested this change specifically?

Reply:
Yes, without this position control is still performed. It is really required to turn of pos control like this since it is turned on by default. Another location would be fine for me as well, but it is important that it is only turned of like this when: 
* altitude control in offboard is turned on
* the ignore flag for pos control is received from the mavlink message

Reply:
Ok good.

Reply:
I found the problem. in the mc_att_controller, the attitude setpoints were only published when either position control, velocity control or acceleration control were running. I added now a condition for when altitude control is running. Now in the commander the velocity control does not need to be set anymore.

Reply:
If you don't reset the velocity setpoints in xy, then the velocity controller will track the previous velocities. 
https://github.com/PX4/Firmware/blob/master/src/modules/mc_pos_control/mc_pos_control_main.cpp#L2362-L2375